### PR TITLE
Move Emotion and MDX packages to peerDeps

### LINKED ISSUE #1388: use peerDependencies to avoid context provider version clash
see https://github.com/system-ui/theme-ui/pull/1351

I claim we can't have MDX packages nor Emotion in dependencies, and we should use peerDependencies instead.

### PR DESCRIPTION
**Pull request by hasparus**
Having Emotion and MDX packages in dependencies leads to a bunch of problems that repeatedly popped up over last few years.

In general, if something depends on a React context, we can't really hide it as dependency from the user, as a version conflict can lead to a context mismatch (we pass the theme to context, but it's read from another one).

We install Emotion for the user, but the abstraction leaks — without knowing that we use Emotion these context mismatch errors  can be absurdly hard to debug.

**Changes in this PR:**

- [x] Move `@emotion/react` and `@emotion/styled` to peer deps.
- [x] Move MDX packages to peer deps.
- [x] Adapt all `npm install` guildelines to include peerDeps.
- [x] Write [release notes](https://intuit.github.io/auto/docs/generated/changelog#additional-release-notes) describing the breaking change.

**Connected issues:**

- https://github.com/system-ui/theme-ui/issues/1130
- https://github.com/system-ui/theme-ui/issues/1793
- https://github.com/system-ui/theme-ui/issues/1531
- https://github.com/system-ui/theme-ui/issues/1530
- This PR closes #1388.
- https://github.com/system-ui/theme-ui/issues/1345#issuecomment-742225675

## Release notes

**BREAKING CHANGE:** `@emotion/react` and `@mdx-js/react`  were moved to peerDependencies of `theme-ui`, `@theme-ui/mdx` and `@theme-ui/sidenav` and must be installed separately. This is a revert of change from Theme UI v0.3, and it's meant to help solve version clashes and context mismatch bugs on user side. Connected issues: [#1793](https://github.com/system-ui/theme-ui/issues/1793), [#1531](https://github.com/system-ui/theme-ui/issues/1531), [#1530](https://github.com/system-ui/theme-ui/issues/1530), [#1388](https://github.com/system-ui/theme-ui/issues/1388), [#1345](https://github.com/system-ui/theme-ui/issues/1345#issuecomment-742225675), [#1130](https://github.com/system-ui/theme-ui/issues/1130).

---

<details>
  <summary>📦 Published PR as canary version: <code>0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @theme-ui/color-modes@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  npm install @theme-ui/color@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  npm install @theme-ui/components@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  npm install @theme-ui/core@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  npm install @theme-ui/css@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  npm install @theme-ui/custom-properties@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  npm install @theme-ui/editor@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  npm install gatsby-plugin-theme-ui@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  npm install gatsby-theme-style-guide@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  npm install gatsby-theme-ui-layout@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  npm install @theme-ui/match-media@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  npm install @theme-ui/mdx@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  npm install @theme-ui/parse-props@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  npm install @theme-ui/preset-base@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  npm install @theme-ui/preset-bootstrap@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  npm install @theme-ui/preset-bulma@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  npm install @theme-ui/preset-dark@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  npm install @theme-ui/preset-deep@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  npm install @theme-ui/preset-funk@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  npm install @theme-ui/preset-future@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  npm install @theme-ui/preset-polaris@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  npm install @theme-ui/preset-roboto@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  npm install @theme-ui/preset-sketchy@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  npm install @theme-ui/preset-swiss@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  npm install @theme-ui/preset-system@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  npm install @theme-ui/preset-tailwind@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  npm install @theme-ui/preset-tosh@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  npm install @theme-ui/presets@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  npm install @theme-ui/prism@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  npm install @theme-ui/sidenav@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  npm install @theme-ui/style-guide@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  npm install @theme-ui/tailwind@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  npm install @theme-ui/theme-provider@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  npm install theme-ui@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  npm install @theme-ui/typography@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  # or 
  yarn add @theme-ui/color-modes@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  yarn add @theme-ui/color@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  yarn add @theme-ui/components@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  yarn add @theme-ui/core@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  yarn add @theme-ui/css@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  yarn add @theme-ui/custom-properties@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  yarn add @theme-ui/editor@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  yarn add gatsby-plugin-theme-ui@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  yarn add gatsby-theme-style-guide@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  yarn add gatsby-theme-ui-layout@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  yarn add @theme-ui/match-media@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  yarn add @theme-ui/mdx@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  yarn add @theme-ui/parse-props@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  yarn add @theme-ui/preset-base@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  yarn add @theme-ui/preset-bootstrap@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  yarn add @theme-ui/preset-bulma@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  yarn add @theme-ui/preset-dark@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  yarn add @theme-ui/preset-deep@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  yarn add @theme-ui/preset-funk@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  yarn add @theme-ui/preset-future@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  yarn add @theme-ui/preset-polaris@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  yarn add @theme-ui/preset-roboto@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  yarn add @theme-ui/preset-sketchy@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  yarn add @theme-ui/preset-swiss@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  yarn add @theme-ui/preset-system@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  yarn add @theme-ui/preset-tailwind@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  yarn add @theme-ui/preset-tosh@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  yarn add @theme-ui/presets@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  yarn add @theme-ui/prism@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  yarn add @theme-ui/sidenav@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  yarn add @theme-ui/style-guide@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  yarn add @theme-ui/tailwind@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  yarn add @theme-ui/theme-provider@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  yarn add theme-ui@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  yarn add @theme-ui/typography@0.13.0--canary.1867.408bee0d2d84fbaf22e6f2ca925d50f4c82ad1db.0
  ```
</details>

### CODE REVIEW

**Comment by Swizec** [comment]
Any prediction when this is getting merged? We keep tripping over it every few months when someone tries to upgrade packages.

File: packages/core/package.json [hunk 1]
```diff
@@ -15,12 +15,11 @@
   },
   "dependencies": {
     "@theme-ui/css": "0.11.3",
-    "@theme-ui/parse-props": "0.11.3",
-    "@emotion/react": "^11.4.1",
-    "deepmerge": "^4.2.2"
```
Comment (flo-sch) [inline packages/core/package.json:20]: Deepmerge as well? Not a typo?

File: packages/parse-props/package.json [hunk 2]
```diff
@@ -18,11 +18,11 @@
     "access": "public"
   },
   "dependencies": {
-    "@theme-ui/css": "0.11.3",
-    "@emotion/react": "^11.4.1"
```
Comment (flo-sch) [inline packages/parse-props/package.json]: Will the `import { Interpolation } from '@emotion/react';` still work if we remove it completely (without adding it to peerDeps)?

**Review by flo-sch — COMMENTED** [review]
Just some imports I am not fully sure about

File: packages/parse-props/package.json (hunk 2 above)
Comment (hasparus) [reply packages/parse-props/package.json]: Actually, we should add an ESLint rule to catch this.

**Comment by hasparus** [comment]
@Swizec, I'll reluctantly say _soon™_. It's been a while since somebody messaged about this problem so I hoped other libs updated to Emotion 11 and it's not occuring so often 😓

**Comment by Swizec** [comment]
@hasparus we recently ran into it _because_ everything updated to latest. But it was like emotion 11.6.0 in one place and 11.5.0 in another 💩

Made up example version numbers but something like that. We had to carefully ensure everything used the exact same version and it was a pain to figure out what the problem was.

File: packages/core/package.json (hunk 1 above)
Comment (hasparus) [reply packages/core/package.json:20]: Definitely a typo. Fixed it.

File: packages/parse-props/package.json (hunk 2 above)
Comment (hasparus) [reply packages/parse-props/package.json]: Added `import/no-extraneous-dependencies`.

**Review by beerose — APPROVED** [review]
(was asked by @hasparus to take a look at this PR)

LGTM

**Comment by hasparus** [comment]
:rocket: PR was released in [`v0.13.0`](https://github.com/system-ui/theme-ui/releases/tag/v0.13.0) :rocket:

**Pull request merged**